src/sink/prometheus.rs

```diff
@@ -377,152 +404,156 @@ impl Prometheus {
     /// Create a new prometheus sink
     ///
     /// Please see documentation on `PrometheusConfig` for more details.
-    pub fn new(config: &PrometheusConfig) -> Prometheus {
+    pub fn new(config: &PrometheusConfig) -> Self {
         let aggrs = PrometheusAggr::new(config.capacity_in_seconds);
         let thrd_aggrs = sync::Arc::new(sync::Mutex::new(None));
         let srv_aggrs = sync::Arc::clone(&thrd_aggrs);
-        let listener = Server::http((config.host.as_str(), config.port))
-            .unwrap()
-            .handle_threads(SenderHandler { aggr: srv_aggrs }, 1)
-            .unwrap();
 
+        let host_port =
```

Comment:
A tuple here should be able to coerce into the IP needed. Yeah, looks like `(&str, u16)` implements [ToSocketAddrs](https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html).